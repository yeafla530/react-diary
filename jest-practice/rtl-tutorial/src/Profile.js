export default function Profile({user}) {
    return (
        <div>
            {user?.name ? (<h1>{user.name}님 환영합니다</h1>) : (
                <h1>
                    로그인을 해주세요
                    <button>로그인</button>
                </h1>
            )}
        </div>
    )
}

