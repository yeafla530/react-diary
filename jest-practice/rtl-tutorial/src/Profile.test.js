import {render, screen} from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import Profile from "./Profile"

test("유저가 없으면 로그인 문구와 버튼을 보여준다", () => {
    render(<Profile />)
    const txtEl = screen.getByText(/로그인을 해주세요/);
    const btnEl = screen.getByRole("button");
    expect(txtEl).toBeInTheDocument();
    expect(btnEl).toBeInTheDocument();
    // button에 로그인 text를 가지고 있어야함
    expect(btnEl).toHaveTextContent("로그인")
})

test("유저가 있으면 환영문구를 보여준다", () => {
    render(<Profile user={{name: "lee"}}/>)
    const txtEl = screen.getByText(/lee님 환영합니다/);
    expect(txtEl).toBeInTheDocument();
})

test("유저가 name이 없으면 로그인 문구와 버튼을 보여준다", () => {
    render(<Profile user="Park"/>)
    const txtEl = screen.getByText(/로그인을 해주세요/);
    const btnEl = screen.getByRole("button");
    expect(txtEl).toBeInTheDocument();
    expect(btnEl).toBeInTheDocument();
    // button에 로그인 text를 가지고 있어야함
    expect(btnEl).toHaveTextContent("로그인")
})