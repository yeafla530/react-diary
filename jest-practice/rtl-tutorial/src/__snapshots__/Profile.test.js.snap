// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<div>
  <div>
    <b>
      velopert
    </b>
     
    <span>
      (
      이예림
      )
    </span>
  </div>
</div>
`;
